### PR TITLE
Replace std::sqrt with sycl::sqrt for SYCL compatibility

### PR DESCRIPTION
**Pull request by orrangetabby17**
This pull request systematically replaces all uses of the standard `std::sqrt` and related math functions with their SYCL equivalents (`sycl::sqrt`) across the XPU SYCL backend. This change ensures that all square root operations are compatible with SYCL device execution, which is essential for portability and performance on supported hardware.

The most important changes include:

**BatchNorm and Math Kernel Updates:**
- Replaced all instances of `std::sqrt` with `sycl::sqrt` in batch normalization kernel functors and supporting math utilities in `BatchNormKernels.cpp`, ensuring device compatibility for normalization calculations.

**Math Extensions and Special Functions:**
- Replaced all `std::sqrt` calls with `sycl::sqrt` in mathematical special functions and helpers, such as incomplete gamma, Bessel, and Airy functions in `MathExtensions.h`, ensuring device execution for these advanced math routines.

** ASM Difference **
ASM analysis shows that `sycl::sqrt` does not utilize the same hardware-native fast instruction path as `std::sqrt`.

### CODE REVIEW

**Comment by orrangetabby17** [comment]
@copilot is UT fail in ci related to this pr?

**Review by jianyizh — CHANGES_REQUESTED** [review]
You need add build flag -ffp32-prec-sqrt and -ffp32-prec-div

**Comment by orrangetabby17** [comment]
@copilot  is UT fail in ci related to this pr?

**Review by jianyizh — APPROVED** [review]

**Comment by jianyizh** [comment]
Some background, previous -cl-fp32-correctly-rounded-divide-sqrt does not work as expected, we use these two flags to do correct rounding.

File: cmake/BuildFlags.cmake [hunk 1]
```diff
@@ -155,7 +155,8 @@ macro(set_build_flags)
 
     set(SYCL_OFFLINE_COMPILER_CG_OPTIONS "${SYCL_OFFLINE_COMPILER_CG_OPTIONS} -options -cl-poison-unsupported-fp64-kernels")
     set(SYCL_OFFLINE_COMPILER_CG_OPTIONS "${SYCL_OFFLINE_COMPILER_CG_OPTIONS} -options -cl-intel-enable-auto-large-GRF-mode")
-    set(SYCL_OFFLINE_COMPILER_CG_OPTIONS "${SYCL_OFFLINE_COMPILER_CG_OPTIONS} -options -cl-fp32-correctly-rounded-divide-sqrt")
+    set(SYCL_OFFLINE_COMPILER_CG_OPTIONS "${SYCL_OFFLINE_COMPILER_CG_OPTIONS} -options -ffp32-prec-sqrt")
+    set(SYCL_OFFLINE_COMPILER_CG_OPTIONS "${SYCL_OFFLINE_COMPILER_CG_OPTIONS} -options -ffp32-prec-div")
```
Comment (CuiYifeng) [inline cmake/BuildFlags.cmake]: Since build options have been updated, please check Windows build at least.

**Review by CuiYifeng — APPROVED** [review]

**Comment by orrangetabby17** [comment]
@copilot is UT fail in ci related to this pr?

**Comment by orrangetabby17** [comment]
@copilot is UT fail in ci related to this pr?
